src/test/kotlin/ecommerce/controller/AuthControllerTest.kt

```diff
@@ -0,0 +1,292 @@
+package ecommerce.controller
+
+import ecommerce.auth.JwtTokenProvider
+import ecommerce.controller.api.AuthController
+import ecommerce.dao.JdbcMemberDAO
+import ecommerce.dto.AuthResponse
+import ecommerce.dto.LoginForm
+import ecommerce.dto.MemberResponse
+import ecommerce.dto.RegisterForm
+import ecommerce.model.Member
+import ecommerce.service.AuthService
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.jdbc.core.JdbcTemplate
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class AuthControllerTest {
+    @Autowired private lateinit var jdbcTemplate: JdbcTemplate
+
+    @Autowired private lateinit var jdbcMemberDAO: JdbcMemberDAO
+
+    @Autowired private lateinit var jwtTokenProvider: JwtTokenProvider
+    private lateinit var authService: AuthService
+    private lateinit var controller: AuthController
```

Comment:
```suggestion
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
class AuthControllerTest {
    @Autowired private lateinit var jdbcTemplate: JdbcTemplate

    @Autowired private lateinit var jdbcMemberDAO: JdbcMemberDAO

    @Autowired private lateinit var jwtTokenProvider: JwtTokenProvider
    @Autowired private lateinit var authService: AuthService
    @Autowired private lateinit var controller: AuthController
```
Since we're using `@SpringBootTest`, why don't we inject services and controllers as well?

To go further, if we have test coverage for the controller using `RestAssured`, do we need another test for `AuthController`? This is not a right or wrong answer, so feel free to tell me your personal opinion :)

In my point of view, `registerMember()` test is a down graded version of `registerMember() - should insert member and return 201 when form is valid`. We could have both, but as the project gets bigger and more tests get created, I think having both for all controllers will result in more time spent creating and managing them.

Still, I want to point out that you did very well on creating fine-grained tests 👍

Reply:
This also leads to another question : `Which tests do you think are necessary when maintaining projects?`

Reply:
> ... we have test coverage for the controller using `RestAssured`, do we need another test for `AuthController`?

The reason was that, I actually not know well and not sure about `RestAssured`. I couldn't guarantee what if, how it work in different test environment depends on system setting. So I tried to have both test cases using `RestAssured` and just test library for my own safety.


> This also leads to another question : `Which tests do you think are necessary when maintaining projects?`

I think it depends on situation and size of an application. However, End-to-End test for an application is must, and other cases are needed to guarantee the functionality of logic, including inputs and outputs of objects and methods.